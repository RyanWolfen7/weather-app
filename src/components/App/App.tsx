import React from 'react';
import { Navbar, NavbarBrand } from 'reactstrap';
import styled from 'styled-components';
import Search from '../Search/Search';

const App: React.FC = ({ children }) => {
  return (
    <>
      <Nav dark>
        <NavbarBrand href={'/'}>{'Weather App'}</NavbarBrand>
        <Search />
      </Nav>

      <Wrapper>
        {children}
      </Wrapper>
    </>
  );
}

export default App;

const Wrapper = styled.div`
  background-image: url(/assets/pattern.svg);
`;

const Nav = styled(Navbar)`
  background: rgba(252,90,44,.89);
  flex-direction: row;
  display: flex;
  justify-content: space-between;
`;
