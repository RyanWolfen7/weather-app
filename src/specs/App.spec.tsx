import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import App from '../components/App/App';

let app: ShallowWrapper;

beforeAll(() => {
  app = shallow(<App />);
});

test('App renders and has the title Weather App', () => {
  expect(app.find('NavbarBrand').children().text()).toEqual('Weather App');
});

test('App to match snapshot', () => {
  expect(app).toMatchSnapshot();
});
