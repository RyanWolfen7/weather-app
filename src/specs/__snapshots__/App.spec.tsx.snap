// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App to match snapshot 1`] = `
<Fragment>
  <Styled(Navbar)
    dark={true}
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      Weather App
    </NavbarBrand>
    <withRouter(Search) />
  </Styled(Navbar)>
  <styled.div />
</Fragment>
`;
